### PR TITLE
Allow specs to vote on votesplitplayers and votemove

### PR DESCRIPTION
**Pull request by sprunk**
Also gives specs a response "only players can vote" if they try to vote on things they can't vote for (like kicks, map etc)

### CODE REVIEW

File: Springie/Springie/autohost/Polls/VoteMap.cs [hunk 1]
```diff
@@ -112,5 +112,17 @@ protected override void SuccessAction()
                 ah.ComMap(TasSayEventArgs.Default, new string[] { map });
             }
         }
+
+         protected override bool AllowVote(TasSayEventArgs e)
```
Comment (Licho1) [inline Springie/Springie/autohost/Polls/VoteMap.cs]: Code quadriplication is ugly, perhaps add this to some abstract class from which these polls inherit?

**Comment by sprunk** [comment]
Base class AllowVote returns true by default. If the no-spec-check was added to the base class then all the classes that allow specs would have to override instead.

**Comment by Licho1** [comment]
You create class that inherits from base, and overrides this method. Then make those polls where vote is allowed inherit from this class instead of original base class.

**Comment by sprunk** [comment]
Oh ok.

**Pull request closed without merging**